### PR TITLE
[Feature] Add inventory (collection) and vending machine

### LINKED ISSUE #49: Vending machine
A place where you can sell stuff (I think)

### PR DESCRIPTION
**Pull request by kida31**
Adds a collection like class for inventory, itemstack (and respective interfaces)
Adds vending machine (backend and ui)

Closes #49

### CODE REVIEW

File: VendingMachine/UI/Tooltip.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
+using Godot;
+
+namespace untitledplantgame.VendingMachine;
+
+public partial class Tooltip : Control
```
Comment (felix-schindler) [inline VendingMachine/UI/Tooltip.cs:5]: Name könnte klarer sein (nur für Gesichter; kein allgemeiner Tooltip)

**Review by felix-schindler — CHANGES_REQUESTED** [review]

File: Inventory/Inventory.cs [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+	private ItemStack AddItem(ItemStack item)
+	{
+		if (item == null)
+		{
+			return null;
+		}
+
+		item = item.Clone() as ItemStack;
+
+		while (true)
```
[the remark this answers is not in the document]
Comment (kida31) [reply Inventory/Inventory.cs]: Rewritten!

File: Inventory/ItemCategory.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
+﻿using System;
+
+namespace untitledplantgame.Inventory;
+
+public class ItemCategory : IEquatable<ItemCategory>
```
[the remark this answers is not in the document]
Comment (kida31) [reply Inventory/ItemCategory.cs]: Added comment to clarify

File: Inventory/IInventory.cs [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+{
+	/// <summary>
+	///     Returns the size of the inventory
+	/// </summary>
+	public int Size { get; }
+
+	/// <summary>
+	///     Returns the maximum stack size for an ItemStack in this inventory.
+	/// </summary>
+	[Obsolete]
```
[the remark this answers is not in the document]
Comment (kida31) [reply Inventory/IInventory.cs]: Removed

File: VendingMachine/UI/Tooltip.cs (hunk 1 above)
Comment (kida31) [reply VendingMachine/UI/Tooltip.cs]: Changed to EmojiTooltip

**Pull request merged**